openbrews/recipe/RecipeUtils.js

```diff
@@ -0,0 +1,131 @@
+angular.module('openbrews.recipeUtils', [])
+.service('RecipeUtils', function () {
+
+    // calculates IBU given a recipe and OG (original gravity).
+    this.calculateIbu = function (recipe, og) {
+        var hops = recipe.hops.filter(function (hop) {
+            return hop.aa;
+        });
+
+        var gravity = this.boilGravity(recipe, og);
+
+        return hops.reduce(function (acc, hop) {
+            return acc + this.ibuFromHop(hop, recipe.batchSize, gravity) / hops.length;
+        }, 0);
+    }
+
+    this.ibuFromHop = function (hop, batchSize, gravity) {
+        return 1.65 * Math.pow(0.000125, gravity - 1) * ((1 - Math.pow(Math.E, -0.04 * hop.time)) / 4.15) *
+        ((hop.type==="Pellet") ? 1.1 : 1) *
+        (((hop.aa / 100) * this.gramsToOz(hop.grams) * 7490) / this.litersToGallons(batchSize));
+    };
+
+    this.boilGravity = function (recipe, og) {
+        var boilSize = (recipe.boilSizeUnits === 'L')? this.litersToGallons(recipe.boilSize) : recipe.boilSize;
+
+        return ((this.calcBatchSize(recipe) / boilSize) * (og - 1) + 1);
+    };
+
+    this.gramsToOz = function (grams) {
+        return grams * 0.0352739619;
+    };
+
+    this.litersToGallons = function (liters) {
+        return liters * 0.26417;
+    };
+
+    this.kgsToLbs = function (kgs) {
+        return kgs * 2.2046;
+    };
+
+    this.mlToFlOz = function (ml) {
+        return ml * 0.033814;
+    }
+
+    // Calculate the total batch size of the recipe.
+    this.calcBatchSize = function (recipe) {
+        var sum = 0;
+
+        sum += recipe.fermentables.reduce(function (acc, frmtable) {
+            return acc + (frmtable.weightUnits === 'Kg')? this.kgsToLbs(frmtable.weight) : frmtable.weight;
```

Comment:
this is adding lbs to a value where only oz are being added. Is this correct? Does this need to be converted to oz first?